internal/venice-common/src/main/java/com/linkedin/venice/controllerapi/D2ControllerClient.java

```diff
@@ -57,10 +62,23 @@ public D2ControllerClient(
     this.sslFactory = sslFactory;
   }
 
+  /**
+   * Discover leader controller from a D2 client.
+   * @return leader controller url
+   */
   @Override
   protected String discoverLeaderController() {
+    List<D2Client> d2Client = Collections.singletonList(this.d2Client);
```

Comment:
Let's create the object once in the constructor and use it every time we need to. Creating a new List every time is not efficient.